hail/python/test/hail/table/test_table.py

```diff
@@ -374,6 +374,47 @@ def test_multiple_entry_joins(self):
         mt.select_entries(a=mt2[mt.row_idx, mt.col_idx].x,
                           b=mt2[mt.row_idx, mt.col_idx].x)
 
+    def test_multi_way_zip_join(self):
+        d1 = [{"id": 0, "name": "a", "data": 0.0},
+              {"id": 1, "name": "b", "data": 3.14},
+              {"id": 2, "name": "c", "data": 2.78}]
+        d2 = [{"id": 0, "name": "d", "data": 1.1},
+              {"id": 0, "name": "x", "data": 2.2},
+              {"id": 2, "name": "v", "data": 7.89}]
+        d3 = [{"id": 1, "name": "f", "data":  9.99},
+              {"id": 2, "name": "g", "data": -1.0},
+              {"id": 3, "name": "z", "data":  0.01}]
+        s = hl.tstruct(id=hl.tint32, name=hl.tstr, data=hl.tfloat64)
+        ts = [hl.Table.parallelize(r, schema=s, key='id') for r in [d1, d2, d3]]
+        joined = hl.Table._multi_way_zip_join(ts, '__data', '__globals').drop('__globals')
```

Comment:
See comment about adding non-empty globals values to the test.

Reply:
done